tests/unit/test_arcaflow_plugin_opensearch.py

```diff
@@ -34,6 +34,44 @@ def test_serialization():
             )
         )
 
+    def test_convert_to_homogeneous_list(self):
+        test_cases = [
+            ["a", "b", "c"],  # all str
+            ["a", "b", 1],  # One final int to convert to str
+            [1.0, 1, "1"],  # str at end, so upconvert all to str
+            ["1", 1, 1.0],
+            ["1", 1, 1],
+            [1, 1, "1"],
+            [1, 1, 1],
+            [1.0, 1, 1],
+            [1, 1, 1.0],
+        ]
+        # Ensure they're all homogeneous
+        for test_case in test_cases:
+            validate_list_items_homogeous_type(
+                self, opensearch_plugin.convert_to_homogenous_list(test_case)
+            )
+        # Ensure the type matches
+        self.assertEqual(
+            int, type(opensearch_plugin.convert_to_homogenous_list([1, 1, 1])[0])
+        )
+        self.assertEqual(
+            float,
+            type(opensearch_plugin.convert_to_homogenous_list([1, 1, 1.0])[0]),
+        )
+        self.assertEqual(
+            str,
+            type(opensearch_plugin.convert_to_homogenous_list([1, 1.0, "1.0"])[0]),
+        )
+
+
+def validate_list_items_homogeous_type(t, input_list):
+    if len(input_list) == 0:
+        return  # no problem with an empty list
```

Comment:
You don't actually have a test case for a zero-length list -- I recommend either adding one or removing this otherwise useless code.